bundle/src/main/java/com/adobe/acs/commons/redirects/models/RedirectConfiguration.java

```diff
@@ -120,19 +125,22 @@ public RedirectMatch match(String requestPath) {
      *     <li>Match by a regular expression. This is O(N) linear lookup in a list of rules keyed by their regex patterns</li>
      * </ol>
      *
-     * @param requestPath   the request to match
+     * @param resourcePath   the request to match
      * @param contextPrefix the optional context prefix to take into account
+     * @param request the current sling request
      * @return  match or null
      */
-    public RedirectMatch match(String requestPath, String contextPrefix) {
-        String normalizedPath = normalizePath(requestPath);
+    public RedirectMatch match(String resourcePath, String contextPrefix, SlingHttpServletRequest request) {
+        String normalizedPath = normalizePath(resourcePath);
         RedirectMatch match = null;
         RedirectRule rule = getPathRule(normalizedPath, contextPrefix);

```

Comment:
1. when initializing RedirectConfiguration , Request URI rules should be keyed by the source path without normalizing. We should also set a flag if any of the rules is Request URI . Something like this:
```java
    public RedirectConfiguration(Resource resource, String storageSuffix) {
        pathRules = new LinkedHashMap<>();
        patternRules = new LinkedHashMap<>();
        path = resource.getPath();
        name = path.replace("/" + storageSuffix, "");
        Collection<RedirectRule> rules = RedirectFilter.getRules(resource);
        for (RedirectRule rule : rules) {
            if (rule.getRegex() != null) {
                patternRules.put(rule.getRegex(), rule);
            } else {
		// request URI rules are keyed without normalizing
                if(rule.getEvaluateURI()){
                    hasNonRegexRequestURIRules = true;
                    pathRules.put(rule.getSource(), rule);
                } else {
                    pathRules.put(normalizePath(rule.getSource()), rule);
                }
            }
        }

    }
```

The 'match' method should do a second try if the `hasNonRegexRequestURIRules`  flag was set:

```java
        RedirectRule rule = getPathRule(normalizedPath, contextPrefix);
        if(rule == null && hasNonRegexRequestURIRules){
	   // there are request URI rules. Check is any mathes
            String pathToEvaluate = determinePathToEvaluate(normalizedPath, true, request);
            rule = getPathRule(pathToEvaluate, contextPrefix);
        }
```


this way request URI will be properly handled by both regex and non-regex rules.